WebService 2 Assignment/WebServiceUserManagement/src/com/metacube/restwebservice/service/EmployeesService.java

```diff
@@ -0,0 +1,163 @@
+
+	EmployeeFacade employeeFacade;
+
+	// constructor initialize the variables
+	public EmployeesService() {
+		employeeFacade = new EmployeeFacade();
+	}
+
+	/**
+	 * This is used to get the employees List
+	 * 
+	 * @return List of employees
+	 */
+	@GET
+	@Path("/getAll/")
+	@Produces({ MediaType.APPLICATION_XML })
+	public List<Employee> getEmployees() {
+		return employeeFacade.getEmployeeAsList();
+	}
+
+	/*
+	 * @GET
+	 * 
+	 * @Produces(MediaType.TEXT_XML) public List<Employee> getEmployeesAsHtml()
+	 * { return employeeService.getEmployeeAsList(); }
+	 */
+	/**
+	 * This is used as post method to create the employee
+	 * 
+	 * @param id
+	 * @param name
+	 * @param age
+	 * @param servletResponse
+	 * @throws IOException
+	 */
+	@POST
+	@Path("/create/")
+	@Produces(MediaType.TEXT_HTML)
+	@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+	public void createEmployee(@FormParam("id") String id,
+			@FormParam("employeename") String name,
+			@FormParam("employeeage") String age,
+			@Context HttpServletResponse servletResponse) throws IOException {
+		Employee employee = new Employee(id, name, age);
+		employeeFacade.createEmployee(employee);
+		servletResponse.sendRedirect("./getAll");
+	}
+
+	/**
+	 * This method is used to get the employee detail by id
+	 * 
+	 * @param id
+	 *            - employee id
+	 * @return
+	 */
+	@GET
+	@Path("getById/{employee}")
+	public Employee getEmployee(@PathParam("employee") String id) {
+		return employeeFacade.getEmployeeById(id);
+	}
+
+	/**
+	 * This method is getting the input from from to get employee detail by id
+	 * 
+	 * @param id
+	 * @param servletResponse
+	 * @throws IOException
+	 */
+	@POST
+	@Path("getById")
+	@Produces(MediaType.TEXT_HTML)
+	@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+	public void getEmployeeById(@FormParam("id") String id,
+			@Context HttpServletResponse servletResponse) throws IOException {
+		servletResponse.sendRedirect("./getById/" + id);
+	}
+
+	/**
+	 * This method is used to delete the employee by id
+	 * 
+	 * @param id
+	 * @return
+	 */
+	@GET
+	@Path("deleteById/{employee}")
+	public List<Employee> deleteEmployee(@PathParam("employee") String id) {
+		employeeFacade.deleteEmployeeById(id);
+		return employeeFacade.getEmployeeAsList();
+	}
+
+	/**
+	 * This method is used to get the input from the form
+	 * 
+	 * @param id
+	 * @param servletResponse
+	 * @throws IOException
+	 */
+	@POST
+	@Path("deleteById")
+	@Produces(MediaType.TEXT_HTML)
+	@Consumes(MediaType.APPLICATION_FORM_URLENCODED)
+	public void deleteEmployeeById(@FormParam("id") String id,
+			@Context HttpServletResponse servletResponse) throws IOException {
+		servletResponse.sendRedirect("./deleteById/" + id);
+	}
+
+	/**
+	 * This method is used to get the employee detail by name
+	 * 
+	 * @param name
+	 * @return
+	 */
+	@GET
+	@Path("getByName/{employee}")
+	public Employee getEmployeeByName(@PathParam("employee") String name) {
+		return employeeFacade.getEmployeeByName(name);
```

Comment:
I think this should return a list, as there may be several employee with same name.